src/main/scala/io/iohk/ethereum/consensus/atomixraft/AtomixRaftConsensus.scala

```diff
@@ -53,33 +53,60 @@ class AtomixRaftConsensus private(
 
   private[this] final val metrics = new AtomixRaftConsensusMetrics(Metrics.get())
 
-  private[this] def setupMiner(node: Node): Unit = {
-    miner.setOnce {
+  private[this] def startForger(node: Node): Unit = {
+    forgerRef.setOnce {
       val miner = AtomixRaftForger(node)
       miner ! Init
       miner
     }
   }
 
+  private[this] def stopForger(): Unit = forgerRef.kill()
+
   private[this] def onRaftServerStarted(messagingService: ManagedMessagingService): Unit = {
     log.info("Raft server started at " + messagingService.endpoint)
   }
 
-  private[this] def onLeader(): Unit = {
+  private[this] def onLeaderWithMiningEnabled(): Unit = {
     metrics.LeaderEvent.trigger()
     metrics.BecomeLeaderCounter.increment()
 
-    miner ! IAmTheLeader
+    forgerRef ! IAmTheLeader
+  }
+
+  private[this] def onLeaderWithMiningDisabled(): Unit = {
+    log.warn(s"***** Elected ${RaftServer.Role.LEADER} but ${ConsensusConfig.Keys.MiningEnabled}=${config.miningEnabled}")
+
+    raftServer.run { server ⇒
+      val cluster = server.cluster()
+      val leader = cluster.getLeader
+
+      // We have just become the leader but what is the probability of having a re-election and a new leader,
+      // thus kicking out a different leader?
+      log.info(s"***** Cluster leader is: ${leader}, demoting")
+      try leader.demote().join()
```

Comment:
Is there a race condition whereby the node is a leader for a short while and might mine? I presume not but just checking.

Reply:
Yeah, its the second part that I was worried _might_ not be implemented, but it is!